crates/oxc_linter/src/rules/jest/padding_around_test_blocks.rs

```diff
@@ -0,0 +1,306 @@
+use std::ops::Deref;
+
+use oxc_ast::{AstKind, ast::Statement};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::AstNode;
+use oxc_span::{CompactStr, GetSpan, Span};
+
+use crate::{
+    context::LintContext,
+    rule::Rule,
+    utils::{
+        JestGeneralFnKind, ParsedGeneralJestFnCall, PossibleJestNode, parse_general_jest_fn_call,
+    },
+};
+
+fn padding_around_test_blocks_diagnostic(span: Span, name: CompactStr) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("Missing padding before {name} block"))
+        .with_help(format!("Make sure there is an empty new line before the {name} block"))
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PaddingAroundTestBlocks;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule enforces a line of padding before and after 1 or more test/it statements
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// const thing = 123;
+    /// test('foo', () => {});
+    /// test('bar', () => {});
+    /// ```
+    ///
+    /// ```js
+    /// const thing = 123;
+    /// it('foo', () => {});
+    /// it('bar', () => {});
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// const thing = 123;
+    ///
+    /// test('foo', () => {});
+    ///
+    /// test('bar', () => {});
+    /// ```
+    ///
+    /// ```js
+    /// const thing = 123;
+    ///
+    /// it('foo', () => {});
+    ///
+    /// it('bar', () => {});
+    /// ```
+    PaddingAroundTestBlocks,
+    jest,
+    style,
+    fix
+);
+
+impl Rule for PaddingAroundTestBlocks {
+    fn run_on_jest_node<'a, 'c>(
+        &self,
+        jest_node: &PossibleJestNode<'a, 'c>,
+        ctx: &'c LintContext<'a>,
+    ) {
+        let node = jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Some(jest_fn_call) = parse_general_jest_fn_call(call_expr, jest_node, ctx) else {
+            return;
+        };
+        let ParsedGeneralJestFnCall { kind, name, .. } = &jest_fn_call;
+        let kind = kind.to_general().unwrap();
+        if kind != JestGeneralFnKind::Test {
+            return;
+        }
+        let scope_node = ctx.nodes().get_node(ctx.scoping().get_node_id(node.scope_id()));
```

Comment:
i decided to go with getting the node for the scope instead of traversing to the grandparent node since I think this will be more resilient to AST refactorings. might not matter but found this when looking for ways to get the sibling statements and thought this was better.

another consideration was the `Visit` trait which would be more robust by not having to check the `AstKind` (i think), but my thought was that we would be doing traversal that was already cached by these `statements` fields. so this solution compromises robustness in favor of speed.